pkg/yurthub/server/server.go

```diff
@@ -60,15 +62,16 @@ type yurtHubServer struct {
 func NewYurtHubServer(cfg *config.YurtHubConfiguration,
 	certificateMgr interfaces.YurtCertificateManager,
 	proxyHandler http.Handler,
-	rest *rest.RestConfigManager) (Server, error) {
+	rest *rest.RestConfigManager,
+	checker healthchecker.HealthChecker) (Server, error) {
 	hubMux := mux.NewRouter()
-	registerHandlers(hubMux, cfg, certificateMgr)
 	restCfg := rest.GetRestConfig(false)
 	clientSet, err := kubernetes.NewForConfig(restCfg)
```

Comment:
clientSet is created with remoteServers[0], so healthy status of all remoteServers will be checked when ota upgrade is executed. so `isEdgeCloudConnected(checker, servers)` can return true, but clientSet maybe fail.

Reply:
@rambohe-ch  how about using `rest.GetRestConfig(true)` to generate a new rest config to solve this problem?

Reply:
@xavier-hou we can not use `rest.GetRestConfig(true)` to fix this problem, because the remote server is healthy when creating clientSet, but the remote server maybe become unhealthy when upgrading pod on the edge.

Reply:
It looks like we need to create a new ClientSet with healthy server when upgrading pod.